backend/src/modules/doubles-tournament/core/domain/MatchesQueue.ts

```diff
@@ -28,11 +30,19 @@ export class MatchesQueue {
     return [...this.queue];
   }
 
-  withStatusChangedMatch(match: QueuedMatch): MatchesQueue {
-    const queue = this.queuedMatches.filter((queuedMatch) => queuedMatch.matchNumber.raw !== match.matchNumber.raw);
+  withStatusChangedMatch(matchNumber: MatchNumber, status: MatchStatus, tableNumber?: number): MatchesQueue {
+    const queue = this.queuedMatches.map((queuedMatch) => {
+      if (queuedMatch.matchNumber.raw === matchNumber.raw) {
```

Comment:
Ten if dobrze zastapic poprzez filter.

Reply:
Ale wtedy dostanę tylko ten jeden mecz w kolejce, a chcę całą kolejkę tylko z jednym updatem.

Reply:
Prawda, to bez tego :)

Reply:
```
return queuedMatch.matchNumber.raw === matchNumber.raw
        ? {
              ...queuedMatch,
             status: status
             tableNumber: queuedMatch.tableNumber ?? tableNumber
            }
      : queuedMatch
```

Reply:
Już tak właśnie sobie wykminiłam :)

Reply:
tylko kolejność tableNumber raczej inna ;)